### PR TITLE
Fixes for Swift 2.0 b2

### PR DESCRIPTION
**Pull request by bendjones**
Tests are passing locally, please let me know if I missed something. By the way I love Masonry and SnapKit!

### CODE REVIEW

File: Source/Constraint.swift [hunk 1]
```diff
@@ -424,6 +425,7 @@ private extension NSLayoutAttribute {
                 case .Leading: return  (Config.interfaceLayoutDirection == .LeftToRight) ? insets.left : -insets.right
                 case .Trailing: return  (Config.interfaceLayoutDirection == .LeftToRight) ? insets.right : -insets.left
                 case .Width, .Height, .NotAnAttribute: return CGFloat(0)
+                default: return CGFloat(0)
```
Comment (robertjpayne) [inline Source/Constraint.swift]: Are you sure these are required? Seems like there may be new attributes we need to manage? We don't want to miss any new ones.

File: Source/Debugging.swift [hunk 2]
```diff
@@ -167,6 +167,7 @@ private extension NSLayoutAttribute {
         case .CenterX:              return "centerX"
         case .CenterY:              return "centerY"
         case .Baseline:             return "baseline"
+        default:                    return "default"
```
Comment (robertjpayne) [inline Source/Debugging.swift]: And again on this, I'd rather build in support for new cases now

**Comment by robertjpayne** [comment]
@bendjones looks pretty good. Main thing is can we list out any new attributes with fatal errors rather than using a default catch all case? Just so we don't miss any and get weird/undefined behaviour happening.

**Comment by bendjones** [comment]
Yeah let me add those missing cases to the tests quick. Totally agree @robertjpayne

**Comment by bendjones** [comment]
Turns out the missing case is NSLayoutAttribute.FirstBaseline which the docs state as the following:

```
NSLayoutAttributeFirstBaseline
The object’s baseline. For objects with more than one line of text, this is the baseline for the topmost line of text.

Available in OS X v10.11 and later.
```

Not sure how we want to handle that in this case. Suggestions?

**Comment by bendjones** [comment]
@robertjpayne I think the above commit handles that new case in a reasonable way. To be the use would be to offset by the Y or bottom in relation to the FirstBaseline. But if you have a different thought I'd love to hear it. I haven't used this constraint myself so I'm just basing my change on the docs.

File: Source/ConstraintAttributes.swift [hunk 3]
```diff
@@ -157,10 +157,10 @@ internal struct ConstraintAttributes: RawOptionSetType, BooleanType {
     }
 }
 internal func +=(inout left: ConstraintAttributes, right: ConstraintAttributes) {
-    left = (left | right)
+    left = left.intersect(right)
```
Comment (jsslai) [inline Source/ConstraintAttributes.swift]: Should this be left.union(right)? Or replace with left.unionInPlace(right)

Comment (robertjpayne) [inline Source/ConstraintAttributes.swift]: yea I think this is wrong too -- it should be a union not a intersect @bendjones can you fix that?

**Comment by bendjones** [comment]
Ok @jsslai and @robertjpayne fixed that intersect and changed them to inPlace operations.

**Comment by robertjpayne** [comment]
LGTM thanks :+1: merged!

**Pull request merged**